Helm k8s openshift (#4764)
ICTU/quality-time: openshift/helm/README.md

# Helm chart

A Helm chart allows for a simple deployment on a Kubernetes cluster. The Helm chart in this directory is based on the docker-compose.yml and the standard images and settings from ICTU.

> Except for the Route these are all standard kubernetes elements. The Route defines the main entry point for the application on OpenShift. It is added as the last element in the [templates/www-service.yaml](templates/www-service.yaml)

Reviewer: ```suggestion
Except for the Route these are all standard kubernetes elements. The Route defines the main entry point for the application on OpenShift. It is added as the last element in the [templates/www-service.yaml](templates/www-service.yaml).
```

Also in a corporate environment, it is usually required to define credentials for pulling the images. For this a [templates/pull-secret.yaml](templates/pull-secret.yaml) file is provided.

> All templates refer to standard values from [values.yaml](values.yaml) uneless these are overriden using the Helm way of value overrides. (Either by defining seperate values or your own values.yaml file).

Reviewer: ```suggestion
All templates refer to standard values from [values.yaml](values.yaml) unless these are overridden using the Helm way of value overrides. (Either by defining seperate values or your own values.yaml file).
```

> ## Pre-requisites

Reviewer: ```suggestion
## Prerequisites
```

You need the following CLI tools:

+ OC CLI the openshift cli
+ Helm the Helm CLI

Login with the OC CLI and select a project/namespace. Then the helm commands can be used.

## Install

>     helm install quality-time .
>     helm install --set pullsecret=mybase64encodeddockerjson quality-time .
>     helm install --values myoverride-values.yaml quality-time . 
> 

Reviewer: ````suggestion
```console
$ helm install quality-time .
$ helm install --set pullsecret=mybase64encodeddockerjson quality-time .
$ helm install --values myoverride-values.yaml quality-time . 
```
````

## Uninstall

>     helm uninstall quality-time

Reviewer: ````suggestion
```console
$ helm uninstall quality-time
```    
````
